kpatch/kpatch

```diff
@@ -116,29 +124,31 @@ find_core_module() {
 	return 1
 }
 
-core_module_loaded () {
-	grep -q "T kpatch_register" /proc/kallsyms
+core_loaded () {
+	grep -q -e "T klp_register_patch" -e "T kpatch_register" /proc/kallsyms
 }
 
 get_module_name () {
 	echo $(readelf -p .gnu.linkonce.this_module $1 | grep '\[.*\]' | awk '{print $3}')
 }
 
 verify_module_checksum () {
-    modname=$(get_module_name $1)
-    [[ -z $modname ]] && return 1
+	modname=$(get_module_name $1)
+	[[ -z $modname ]] && return 1
 
-    checksum=$(readelf -p .kpatch.checksum $1 | grep '\[.*\]' | awk '{print $3}')
-    [[ -z $checksum ]] && return 1
+	checksum=$(readelf -p .kpatch.checksum $1 | grep '\[.*\]' | awk '{print $3}')
 
-    sysfs_checksum=$(cat /sys/kernel/kpatch/patches/${modname}/checksum)
-    [[ $checksum == $sysfs_checksum ]] || return 1
+	# Fail checksum match only if both exist and diverge
+	if [[ ! -z $checksum ]] && [[ -e "$SYSFS/${modname}/checksum" ]] ; then
+		sysfs_checksum=$(cat $SYSFS/${modname}/checksum)
+		[[ $checksum == $sysfs_checksum ]] || return 1
+	fi
```

Comment:
Bash functions return the return the result of the last command -- in this case, the "if; fi" construct.  It would probably be safer to explicitly return 0 at the end of the function.